xbmc/addons/AddonManager.cpp

```diff
@@ -1072,46 +1028,8 @@ bool CAddonMgr::IsFromOfficialRepo(const AddonPtr& addon) const
 
 bool CAddonMgr::IsFromOfficialRepo(const AddonPtr& addon, bool bCheckAddonPath) const
 {
-  auto comparator = [&](const std::string& officialRepo) {
-    const std::vector<std::string> officialRepoProps = StringUtils::Split(officialRepo, '|');
-
-    if (bCheckAddonPath)
-    {
-      return (addon->Origin() == officialRepoProps.front() &&
-              StringUtils::StartsWithNoCase(addon->Path(), officialRepoProps.back()));
-    }
-
-    return addon->Origin() == officialRepoProps.front();
-  };
-
-  return addon->Origin() == ORIGIN_SYSTEM ||
-         std::any_of(m_officialAddonRepos.begin(), m_officialAddonRepos.end(), comparator);
-}
-
-void CAddonMgr::AddAddonIfLatest(const AddonPtr& addonToAdd,
-                                 std::map<std::string, AddonPtr>& map) const
-{
-  const auto& latestKnown = map.find(addonToAdd->ID());
-  if (latestKnown == map.end() || addonToAdd->Version() > latestKnown->second->Version())
-    map[addonToAdd->ID()] = addonToAdd;
-}
-
-bool CAddonMgr::FindAddonAndCheckForUpdate(const AddonPtr& addonToCheck,
-                                           const std::map<std::string, AddonPtr>& map,
-                                           VECADDONS& vecAddons) const
-{
-  const auto& remote = map.find(addonToCheck->ID());
-  if (remote != map.end()) // is addon in the desired map?
-  {
-    if (remote->second->Version() > addonToCheck->Version())
-    {
-      // queue addon update
-      vecAddons.emplace_back(remote->second);
-    }
-    return true;
-  }
-
-  return false;
+  CAddonRepos RepoInfos;
+  return RepoInfos.IsFromOfficialRepo(addon, bCheckAddonPath);
```

Comment:
is there a nicer way to do this?

Reply:
Yes, I don't see any reason why this is can't be a static call. It would mean loading the official repos statically also which should be ok as the call to static CompileInfo is static.